### PR TITLE
Feat/add support for  _varchar

### PR DESCRIPTION
**Pull request by Jordan-M-Young**
The goal of this PR is to solve the issue: [Add support for _varchar #480](https://github.com/sfu-db/connector-x/issues/480).  This issue centers around connectorx panicking when interacting with tables containing a varchar[] type. 

To solve this, I added:

- a VarcharArray type in the various enums in connectorx/src/sources/postgres/typesystem.rs
- a Utf8Array type in the various enums in connectorx/src/destinations/arrow2/typesystem.rs
- a mapping between VarcharArray and Utf8Array in connectorx/src/transports/postgres_arrow2.rs
- implemented the ArrowAssoc traint for Option<Vec<String>> and Vec<String> in connectorx/src/destinations/arrow2/arrow_assoc.rs
- added _varchar/varchar[] data to scripts/postgres.sql
- added a test for _varchar/varchar[] compatibility with postgres/polars in connectorx/tests/test_polars.rs

Excited to hear feedback, make edits, and hopefully merge this PR into the project! This is my first ever OS contribution so I apologize for any rookie mistakes in advance!

### CODE REVIEW

**Comment by wangxiaoying** [comment]
Thanks @Jordan-M-Young for the PR! It looks good in general, and I left a minor comment. Also started the CI check.

File: connectorx/src/macros.rs
[the remark this answers is not in the document]
Comment (Jordan-M-Young) [reply connectorx/src/macros.rs:1]: agreed, I've removed these changes.

**Comment by Jordan-M-Young** [comment]
Hey @wangxiaoying I see that the testing stage of the CI pipelines failed on this latest check. I ran 

`cargo clean && just test` locally and my test passes. I am using an updated version of the postgres.sql script to populate my dev postgres instance could that be the issue? 

Have a nice weekend!

**Comment by wangxiaoying** [comment]
Looks like there is a syntax error when initializing postgres db [here](https://github.com/sfu-db/connector-x/actions/runs/4701828786/jobs/8343233953?pr=487#step:8:50).

**Comment by wangxiaoying** [comment]
Hi @Jordan-M-Young , yeah I think that's probably the issue. We use postgres official docker hub image in our CI. You can use it to test locally and see whether your insert value sql can be run on it.

Have a nice weekend to you too : )

**Comment by Jordan-M-Young** [comment]
@wangxiaoying You we're right, there was a typo in postgres.sql causing insert operation to fail. Hopefully its fixed now!

**Comment by wangxiaoying** [comment]
Thank you @Jordan-M-Young !

**Comment by auyer** [comment]
This is really cool! Thanks @Jordan-M-Young !
@wangxiaoying , do you mind generating a new Release, or rc? :smile:

**Comment by Jordan-M-Young** [comment]
Thanks for your help with this @wangxiaoying ! Looking forward to contributing more in the future.

**Pull request merged**